common/flatpak-dir.c

```diff
@@ -7315,6 +7332,9 @@ flatpak_dir_deploy (FlatpakDir          *self,
   if (!flatpak_dir_set_active (self, ref, checkout_basename, cancellable, error))
     return FALSE;
 
+  if (!flatpak_dir_update_deploy_ref (self, ref, checksum, error))
```

Comment:
It's too bad that if the operation is interrupted (e.g. with `kill -9`) you end up with a deploy without a deploy ref. I'm not sure how we could make it atomic, but I wonder if `flatpak repair` should check for missing deploy refs?

Reply:
Yeah, that would be nice. Not critical though. I'll file an issue.